document/document_test.go

```diff
@@ -402,29 +405,31 @@ func TestNewFromStruct(t *testing.T) {
 			case 14:
 				require.EqualValues(t, document.DocumentValue, v.Type)
 			case 15:
-				require.EqualValues(t, document.NullValue, v.Type)
```

Comment:
Could you explain why this has changed? This means that empty pointers are transformed to integers?